gammapy/estimators/map_profile.py

```diff
@@ -0,0 +1,313 @@
+                    spds.counts.data[mask].sum(),
+                    spds.background.data[mask].sum(),
+                )
+
+            result = {
+                "x_min": self.axis.edges[index],
+                "x_max": self.axis.edges[index+1],
+                "x_ref": self.axis.center[index]
+            }
+            if isinstance(spds, SpectrumDatasetOnOff):
+                result["alpha"] = stats.alpha
+            result.update({
+                "counts": stats.n_on,
+                "background": stats.background,
+                "excess": stats.excess,
+            })
+
+            if "ts" in steps:
+                if isinstance(stats.delta_ts, list) or isinstance(stats.delta_ts, np.ndarray):
+                    result["ts"] = np.asanyarray(stats.delta_ts[0],dtype=np.float32)
+                else:
+                    result["ts"] = np.asanyarray(stats.delta_ts, dtype=np.float32)
+                if isinstance(stats.significance, list) or isinstance(stats.significance, np.ndarray):
+                    result["sqrt_ts"] = np.asanyarray(stats.significance[0], dtype=np.float32)
+                else:
+                    result["sqrt_ts"] = np.asanyarray(stats.significance, dtype=np.float32)
+
+            if "err" in steps:
+                result["err"] = np.asanyarray(stats.error, dtype=np.float32)
+
+            if "errn-errp" in steps:
+                result["errn"] = stats.compute_errn(self.n_sigma)
+                #ToDo: check the type of errp compared to errn (maybe not the same)
+                _errp = stats.compute_errp(self.n_sigma)
+                if isinstance(_errp, list) or isinstance(_errp, np.ndarray):
+                    result["errp"] = np.asanyarray(_errp[0], dtype=np.float32)
+                else:
+                    result["errp"] = np.asanyarray(_errp, dtype=np.float32)
+ 
+            if "ul" in steps:
+                result["ul"] = stats.compute_upper_limit(self.n_sigma_ul)
+ 
+            # What exposure to take? And waht about the mask?
+            result.update(
+                {"exposure": spds.aeff.data.data.max() * spds.livetime,
+                 "solid_angle": spds.counts.geom.solid_angle()}
+            )
+            results.append(result)
+        return results
+
+    def run(self, dataset, steps="all"):
+        """Make the profiles
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset` or `~gammapy.datasets.MapDatasetOnOff`
+            the dataset to use for profile extraction
+        steps : list of str
+            the steps to be used.
+
+        Returns
+        --------
+        imageprofile : `~gammapy.estimators.ImageProfile`
+            Return an image profile class containing the result
+        """
+        results = self.make_prof(dataset, steps)
+        table = table_from_row_data(results)
+        if isinstance(self.regions[0], RectangleSkyRegion):
+            table.meta["PROFILE_TYPE"] = "orthogonal"
```

Comment:
If we introduce this kind of meta information it needs to be documented somewhere, e.g. what are the other profile types and how are they defined?

Reply:
I put this metadata to anticipate the development of other type of projection (e.g. profile within wedges, profile within annulus, etc). This metadata can then be used in the used to identify which type of projection is used (using indeed the document of this class).
For the moment, I think that this metadata is not super useful as only one projection method is used. But, I propose to keep it for the future...

Reply:
Then maybe put one sentence in the docstring to explain what `orthogonal` means? I found it not obvious. I guess other profile types would be `radial` or similar? I think with the concept of using regions (which I like a lot!) we somehow gave up on simple definitions of profiles such as "along x", "along y" or "radial". In that sense it's probably again the best solution to just keep the region information in the result table. Then the information would be complete...

Reply:
I fully agree that this metadata deserves an explanation...

We have thought with @registerrier to store the regions in the final results of the profiles. This can be the solution... the class `ImageProfile` having 2 arguments, the table and the list of regions, by removing then this metadata.

Just tell me your preference and I implement it

Reply:
The dictionnary metadata and their keyword are described within the documentation